File: iosApp/iosAppTests/swap/TestSwapService.swift
//
//  TestSwapService.swift
//  iosAppTests
//
//  Created by jimmyt on 8/21/22.
//  Copyright © 2022 orgName. All rights reserved.
//

import BigInt
import Foundation
@testable import iosApp

/**
 A basic `SwapNotifiable` implementation used to satisfy `OfferService`'s `swapService` dependency for testing non-swap-related code.
 */
class TestSwapService: SwapNotifiable {
    /**
     Does nothing, required to adopt `SwapNotifiable`. Should not be used.
     */
    func sendTakerInformationMessage(swapID: UUID, chainID: BigUInt) throws -> Bool { return false }
    /**
     Does nothing, required to adopt `SwapNotifiable`. Should not be used.
     */
    func handleFailedTransaction(_ transaction: BlockchainTransaction, error: BlockchainTransactionError) throws {}
    /**
     Does nothing, required to adopt `SwapNotifiable`. Should not be used.
     */
    func handleNewSwap(takenOffer: Offer) throws {}
    /**
     Does nothing, required to adopt `OfferNotifiable`. Should not be used.
     */
    func handleTokenTransferApprovalEvent(_ event: ApprovalEvent) {}
    /**
     Does nothing, required to adopt `SwapNotifiable`. Should not be used.
     */
    func handleSwapFilledEvent(_ event: SwapFilledEvent) throws {}
    /**
     Does nothing, required to adopt `SwapNotifiable`. Should not be used.
     */
    func handlePaymentSentEvent(_ event: PaymentSentEvent) throws {}
    /**
     Does nothing, required to adopt `SwapNotifiable`. Should not be used.
     */
    func handlePaymentReceivedEvent(_ event: PaymentReceivedEvent) throws {}
    /**
     Does nothing, required to adopt `SwapNotifiable`. Should not be used.
     */
    func handleBuyerClosedEvent(_ event: BuyerClosedEvent) throws {}
    /**
     Does nothing, required to adopt `SwapNotifiable`. Should not be used.
     */
    func handleSellerClosedEvent(_ event: SellerClosedEvent) throws {}
}

File: androidApp/src/test/java/com/commuto/interfacemobile/android/swap/TestSwapService.kt
package com.commuto.interfacemobile.android.swap

import com.commuto.interfacemobile.android.blockchain.BlockchainTransaction
import com.commuto.interfacemobile.android.blockchain.BlockchainTransactionException
import com.commuto.interfacemobile.android.blockchain.events.commutoswap.*
import com.commuto.interfacemobile.android.offer.Offer
import com.commuto.interfacemobile.android.offer.OfferService
import java.math.BigInteger
import java.util.*

/**
 * A basic [SwapNotifiable] implementation used to satisfy [OfferService]'s swapService dependency for testing
 * non-swap-related code.
 */
class TestSwapService: SwapNotifiable {
    /**
     * Does nothing, required to adopt [SwapNotifiable]. Should not be used.
     */
    override suspend fun sendTakerInformationMessage(swapID: UUID, chainID: BigInteger): Boolean { return false }
    /**
     * Does nothing, required to adopt [SwapNotifiable]. Should not be used.
     */
    override suspend fun handleFailedTransaction(
        transaction: BlockchainTransaction,
        exception: BlockchainTransactionException
    ) {}
    /**
     * Does nothing, required to adopt [SwapNotifiable]. Should not be used.
     */
    override suspend fun handleNewSwap(takenOffer: Offer) {}
    /**
     * Does nothing, required to adopt [SwapNotifiable]. Should not be used.
     */
    override suspend fun handleSwapFilledEvent(event: SwapFilledEvent) {}
    /**
     * Does nothing, required to adopt [SwapNotifiable]. Should not be used.
     */
    override suspend fun handlePaymentSentEvent(event: PaymentSentEvent) {}
    /**
     * Does nothing, required to adopt [SwapNotifiable]. Should not be used.
     */
    override suspend fun handlePaymentReceivedEvent(event: PaymentReceivedEvent) {}
    /**
     * Does nothing, required to adopt [SwapNotifiable]. Should not be used.
     */
    override suspend fun handleBuyerClosedEvent(event: BuyerClosedEvent) {}
    /**
     * Does nothing, required to adopt [SwapNotifiable]. Should not be used.
     */
    override suspend fun handleSellerClosedEvent(event: SellerClosedEvent) {}
}